diff -U 3 -H -w -E -d -r -N -- chyrp/.gitignore News/.gitignore
--- chyrp/.gitignore	1970-01-01 01:00:00.000000000 +0100
+++ News/.gitignore	2011-03-31 20:32:51.000000000 +0200
@@ -0,0 +1,8 @@
+.DS_Store
+.htaccess
+feathers/photo/lib/cache/*
+uploads/*
+includes/config.yaml.php
+includes/database.yaml.php
+themes/*
+!themes/stardust
diff -U 3 -H -w -E -d -r -N -- chyrp/.htaccess News/.htaccess
--- chyrp/.htaccess	1970-01-01 01:00:00.000000000 +0100
+++ News/.htaccess	2013-03-17 00:59:57.000000000 +0100
@@ -0,0 +1,7 @@
+<IfModule mod_rewrite.c>
+RewriteEngine On
+RewriteBase /News/
+RewriteCond %{REQUEST_FILENAME} !-f
+RewriteCond %{REQUEST_FILENAME} !-d
+RewriteRule ^.+$ index.php [L]
+</IfModule>
diff -U 3 -H -w -E -d -r -N -- chyrp/themes/stardust/content/comment.twig News/themes/stardust/content/comment.twig
--- chyrp/themes/stardust/content/comment.twig	2011-03-31 11:32:50.000000000 +0200
+++ News/themes/stardust/content/comment.twig	2013-03-17 02:22:34.000000000 +0100
@@ -2,12 +2,12 @@
             {% if comment.status == "denied" %}
             <span class="waiting">${ "Your comment is awaiting moderation." | translate }</span>
             {% endif %}
+            ${ "<strong>%s</strong> on <a href=\"%s\" class=\"permalink\">%s</a>" | translate | format(comment.author_link, comment.post.url ~ "#comment_"~comment.id, comment.created_at | strftime("%B %d, %Y, at %I:%M %p" | translate)) }
             <blockquote>
                 $comment.body
             </blockquote>
             <cite>
                 ${ comment.delete_link("Delete" | translate, "") }
                 ${ comment.edit_link("Edit" | translate, "") }
-                ${ "<strong>%s</strong> on <a href=\"%s\" class=\"permalink\">%s</a>" | translate | format(comment.author_link, comment.post.url ~ "#comment_"~comment.id, comment.created_at | strftime("%B %d, %Y, at %I:%M %p" | translate)) }
             </cite>
         </li>
diff -U 3 -H -w -E -d -r -N -- chyrp/themes/stardust/content/post.twig News/themes/stardust/content/post.twig
--- chyrp/themes/stardust/content/post.twig	2011-03-31 11:32:50.000000000 +0200
+++ News/themes/stardust/content/post.twig	2013-05-09 13:17:54.000000000 +0200
@@ -1,8 +1,4 @@
-                    {% if post.first_of_day %}
-                    <div class="date">${ post.created_at | strftime("%B %d" | translate) }</div>
-                    {% endif %}
-                    <div class="post $post.feather{% if post.pinned %} pinned{% endif %}" id="post_$post.id">
-                        {% if post.pinned %}<span class="pinned_text">${ "!" | translate }</span>{% endif %}
+                    <div class="post $post.feather" id="post_$post.id">
                         {% block content %}{% endblock %}
                         {% if route.action == "view" and post.paginated and post.paginated.pages > 1 %}
                         <div class="clear"></div>
@@ -11,30 +7,12 @@
                         <span class="pages">${ "Page %d of %d" | translate | format(post.paginated.page, post.paginated.pages) }</span>
                         {% endif %}
                         <div class="clear"></div>
-                        <span class="info">
-                            {% if post.pinned %}
-                            <strong><a href="$post.url">${ post.created_at | strftime("%B %d @ %I:%M %p" | translate) }</a></strong>
-                            {% else %}
-                            <strong><a href="$post.url">${ post.created_at | strftime("%I:%M %p" | translate) }</a></strong>
-                            {% endif %}
-                            {% if enabled_modules.categorize and post.category.id != 0 %}
-                            |
-                            <a href="$post.category.url" rel="tag">$post.category.name</a>
-                            {% endif %}
+                        <p class="info">
                             {% if enabled_modules.comments %}
-                            |
-                            <a href="$post.url#comments">${ "%s Comment" | translate_plural("%s Comments", post.comment_count) | format(post.comment_count) }</a>
-                            {% endif %}
-                            {% if post.linked_tags | length > 0 %}
-                            |
-                            ${ "Tags" | translate }: ${ post.linked_tags | join(", ") }
-                            {% endif %}
-                        </span>
-                        {% if post.editable or post.deletable %}
-                        <span class="controls">
-                            $post.edit_link
-                            ${ post.delete_link("Delete" | translate, " | ") }
-                        </span>
+                            <a href="$post.url#comments">${ "See %s Comment" | translate_plural("See %s Comments", post.comment_count) | format(post.comment_count) }</a>
                         {% endif %}
-                        <div class="clear"></div>
+                        </p>
                     </div>
+                    {% if route.action != "view" %}
+                    <hr/>
+                    {% endif %}
diff -U 3 -H -w -E -d -r -N -- chyrp/themes/stardust/content/sidebar.twig News/themes/stardust/content/sidebar.twig
--- chyrp/themes/stardust/content/sidebar.twig	2011-03-31 11:32:50.000000000 +0200
+++ News/themes/stardust/content/sidebar.twig	2013-03-17 01:18:22.000000000 +0100
@@ -1,4 +1,3 @@
-            <div id="sidebar">
                 <form action="$site.url" method="get" accept-charset="utf-8">
                     <fieldset>
                         <input type="hidden" name="action" value="search" id="action" />
@@ -7,7 +6,7 @@
                 </form>
 
                 {% if theme.pages_list %}
-                <h1>${ "Pages" | translate }</h1>
+                <h2 class="box-header">${ "Pages" | translate }</h2>
                 <ul>
                     {% for item in theme.pages_list %}
                     <li>
@@ -23,13 +22,10 @@
                 </ul>
                 {% endif %}
 
-                <h1>${ "Welcome, %s!" | translate | format(visitor.login | fallback("Guest" | translate)) }</h1>
-                <ul>
+                <h2 class="box-header">${ "Welcome, %s!" | translate | format(visitor.login | fallback("Guest" | translate)) }</h2>
+                <ul class="navigation">
                     {% if visitor.logged_in %}
                     <li><a href="{% url "controls" %}">${ "User Controls" | translate }</a></li>
-                    {% if visitor.group.can("add_post", "add_page", "view_own_draft", "view_draft", "change_settings") %}
-                    <li><a class="toggle_admin" href="{% url "toggle_admin" %}">${ "Toggle Admin Bar" | translate }</a></li>
-                    {% endif %}
                     <li><a href="{% url "logout" %}">${ "Log Out" | translate }</a></li>
                     {% else %}
                     <li><a href="{% url "login" %}">${ "Log In" | translate }</a></li>
@@ -41,32 +37,14 @@
                 </ul>
 
 ${ trigger.call("sidebar") }
-                {% if tags %}
-                <h1>${ "Tags" | translate }</h1>
-                <ul id="tags_list">
-                    {% for tag in tags %}
-                    <li>
-                        <a href="{% url "tag/" ~ tag.url %}" title="${ "%d post tagged &quot;%s&quot;" | translate_plural("%d posts tagged &quot;%s&quot;", tag.popularity) | format(tag.popularity, (tag.name | escape)) }">$tag.name</a>
-                        <span class="sub">($tag.popularity)</span>
-                        {% if route.action == "tag" and not (GET.name | contains(tag.clean)) %}
-                        <span class="sub">&nbsp;<a href="{% url "tag/"~ tag.url ~"+"~ (GET.name | urlencode) %}">(+)</a></span>
-                        {% endif %}
-                    </li>
-                    {% endfor %}
-                    <li><a href="{% url "tags" %}">${ "All &rarr;" | translate }</a></li>
-                </ul>
-                {% endif %}
-
                 {% if theme.archives_list %}
-                <h1>${ "Archives" | translate }</h1>
-                <ul>
+                <h2 class="box-header">${ "Archives" | translate }</h2>
+                <ul class="navigation">
                     {% for archive in theme.archives_list %}
                     <li>
-                        <a href="$archive.url">${ archive.when | strftime("%B %Y" | translate) }</a>
-                        <span class="sub">($archive.count)</span>
+                        <a href="$archive.url">${ archive.when | strftime("%B %Y" | translate) } ($archive.count)</a>
                     </li>
                     {% endfor %}
                     <li><a href="{% url "archive" %}">${ "All &rarr;" | translate }</a></li>
                 </ul>
                 {% endif %}
-            </div>
diff -U 3 -H -w -E -d -r -N -- chyrp/themes/stardust/feathers/text.twig News/themes/stardust/feathers/text.twig
--- chyrp/themes/stardust/feathers/text.twig	2011-03-31 11:32:50.000000000 +0200
+++ News/themes/stardust/feathers/text.twig	2013-03-17 01:51:31.000000000 +0100
@@ -2,9 +2,12 @@
 
 {% block content %}
 {% if post['title'] %}
-                        <h2 class="title">
+                        <p>
+                          <span style="font-size: 20px">&gt;
                             <a href="$post.url" rel="bookmark" title="${ "Permanent Link to" | translate } &quot;$post.title&quot;">$post.title</a>
-                        </h2>
+                          </span><br/>
+                          On <i>${ post.created_at | strftime("%Y-%m-%d") }</i> by<i> falkTX</i>
+                        </p>
 {% endif %}
                         ${ post.body | read_more }
 {% endblock %}
diff -U 3 -H -w -E -d -r -N -- chyrp/themes/stardust/layouts/default.twig News/themes/stardust/layouts/default.twig
--- chyrp/themes/stardust/layouts/default.twig	2011-03-31 11:32:50.000000000 +0200
+++ News/themes/stardust/layouts/default.twig	2014-04-10 01:20:00.000000000 +0200
@@ -1,54 +1,150 @@
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
-    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
-<html xmlns="http://www.w3.org/1999/xhtml">
-    <head profile="http://gmpg.org/xfn/11">
-        <meta http-equiv="Content-type" content="$theme.type; charset=utf-8" />
-        <title>{% block title %}$site.name{% if title %} &raquo; ${ title | escape }{% endif %}{% endblock %}</title>
-        <meta name="description" content="$site.description" />
-        <meta name="generator" content="Chyrp" />
-        <link rel="pingback" href="$site.chyrp_url/includes/xmlrpc.php" />
-        <link rel="EditURI" type="application/rsd+xml" href="$site.chyrp_url/includes/rsd.php" />
-        <link rel="shortcut icon" href="$site.url/favicon.ico" />
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en" dir="ltr">
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+    <!--[if lt IE 7]>
+    <meta http-equiv="imagetoolbar" content="no" />
+    <![endif]-->
+
+    <title>KXStudio News</title>
+
+    <link rel="copyright" href="http://creativecommons.org/licenses/publicdomain/" />
+    <link rel="stylesheet" href="../css/style.fluid.css" type="text/css" media="screen" />
+    <link rel="stylesheet" href="../css/style.css" type="text/css" media="screen" />
+    <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Raleway:200,700,800,400" type="text/css" media="screen" />
+    <link rel="icon" type="image/png" href="../images/ico_kxstudio.png" />
+    <link rel="shortcut icon" href="../favicon.ico" />
         {% if site.clean_urls %}<link rel="canonical" href="{% if (post.url) %}$post.clean{% elseif page.url %}$page.clean {% else %}$site.url{% endif %}" />{% endif %}
         $theme.feeds
         $theme.stylesheets
         $theme.javascripts
 ${ trigger.call("head") }
+    <script type="text/javascript">
+        function closeAllMenus(){
+            var objSoftwareMenu     = document.getElementById("div_menuSoftware");
+            var objRepositoriesMenu = document.getElementById("div_menuRepositories");
+            var objSupportMenu      = document.getElementById("div_menuSupport");
+            var objCommunityMenu    = document.getElementById("div_menuCommunity");
+
+            if (objSoftwareMenu.style.visibility == "visible")
+                objSoftwareMenu.style.visibility = "hidden";
+
+            if (objRepositoriesMenu.style.visibility == "visible")
+                objRepositoriesMenu.style.visibility = "hidden";
+
+            if (objSupportMenu.style.visibility == "visible")
+                objSupportMenu.style.visibility = "hidden";
+
+            if (objCommunityMenu.style.visibility == "visible")
+                objCommunityMenu.style.visibility = "hidden";
+        };
+
+        function menuItemClicked(event, menu, itemName){
+            event.stopPropagation();
+
+            closeAllMenus();
+
+            //if (menu.style.visibility != "visible")
+            {
+                var leftPos = document.getElementById(itemName).getBoundingClientRect().left - 1;
+                menu.style.left       = "" + leftPos + "px";
+                menu.style.visibility = "visible";
+            }
+            //else
+            //{
+            //    menu.style.visibility = "hidden";
+            //}
+
+            return false;
+        };
+
+        function bodyClicked(){
+            closeAllMenus();
+            return true;
+        };
+
+        function itemSoftwareClicked(event)    { return menuItemClicked(event, document.getElementById("div_menuSoftware"),     "item-soft");      };
+        function itemRepositoriesClicked(event){ return menuItemClicked(event, document.getElementById("div_menuRepositories"), "item-repos");     };
+        function itemSupportClicked(event)     { return menuItemClicked(event, document.getElementById("div_menuSupport"),      "item-support");   };
+        function itemCommunityClicked(event)   { return menuItemClicked(event, document.getElementById("div_menuCommunity"),    "item-community"); };
+
+    </script>
     </head>
-    <body>
-        {% if visitor.group.can("add_post", "add_draft", "add_page", "view_own_draft", "view_draft", "change_settings", "toggle_extensions") %}
-        <div id="controls"{% if hide_admin %} style="display: none"{% endif %}>
-            <ul>
-                {% if visitor.group.can("add_post") %}
-                <li><a id="add_post" href="{% admin "write_post" %}">${ "Write" | translate }</a></li>
-                {% elseif visitor.group.can("add_draft") %}
-                <li><a id="add_draft" href="{% admin "write_post" %}">${ "Write Draft" | translate }</a></li>
-                {% endif %}
-                {% if visitor.group.can("add_page") %}
-                <li><a id="add_page" href="{% admin "write_page" %}">${ "Add Page" | translate }</a></li>
-                {% endif %}
-                {% if visitor.group.can("view_own_draft", "view_draft") %}
-                <li><a id="your_drafts" href="{% url "drafts" %}">${ "Your Drafts" | translate }</a></li>
-                {% endif %}
-                {% if visitor.group.can("change_settings", "toggle_extensions") %}
-                <li><a id="site_settings" href="$site.chyrp_url/admin/">${ "Admin" | translate }</a></li>
-                {% endif %}
-                <li class="close"><a class="toggle_admin" href="{% url "toggle_admin" %}">${ "Close" | translate }</a></li>
-            </ul>
+<body class="mediawiki ltr ns-0 ns-subject skin-bentofluid" style="min-width:1080px" onClick="return bodyClicked();">
+    <!-- Software sub menu -->
+    <div id="div_menuSoftware" class="div_menuDropdown">
+        <p><a href="/Applications"><img src="/images/ico_cadence.png" alt=""/> &nbsp; Applications</a></p>
+        <p><a href="/Plugins"><img src="/images/ico_distrho.png" alt=""/> &nbsp; Plugins</a></p>
         </div>
-        {% endif %}
-        {% if site.name %}
+    <!-- Repositories sub menu -->
+    <div id="div_menuRepositories" class="div_menuDropdown">
+        <p><a href="/Repositories">About</a></p>
+        <p><a href="/Repositories:Applications">Applications (in Repo)</a></p>
+        <p><a href="/Repositories:Plugins">Plugins (in Repo)</a></p>
+    </div>
+    <!-- Support sub menu -->
+    <div id="div_menuSupport" class="div_menuDropdown">
+        <p><a href="/Documentation">Documentation</a></p>
+    </div>
+    <!-- Community sub menu -->
+    <div id="div_menuCommunity" class="div_menuDropdown">
+        <p><a href="http://webchat.freenode.net/?channels=#kxstudio,#opensourcemusicians"><img src="/images/ico_chat.png" class="external" rel="nofollow" target="_blank" alt=""/> &nbsp; Chat</a></p>
+        <p><a href="http://linuxmusicians.com/viewforum.php?f=47"><img src="/images/ico_forum.png" class="external" rel="nofollow" target="_blank" alt=""/> &nbsp; Forum</a></p>
+        <p><a href="https://soundcloud.com/groups/kxstudio"><img src="/images/ico_soundcloud.png" class="external" rel="nofollow" target="_blank" alt=""/> &nbsp; Songs</a></p>
+    </div>
+
+    <!-- Top link -->
+    <a name="Top" id="Top"></a>
+
+    <!-- Start: Header -->
         <div id="header">
-            <div class="header_box">
-                <h1><a href="$site.url">$site.name</a></h1>
-                {% if site.description %}
-                <h2>$site.description</h2>
-                {% endif %}
+        <div id="header-content" class="container_12">
+            <a id="header-logo" href="/">
+                <img src="../images/header-logo_mod.png" width="46" height="26" alt="Header Logo"/>
+            </a>
+            <a id="header-home" href="/" style="font-size:1.2em">Home</a>
+            <ul id="global-navigation">
+                <li id="item-news"><a href="/News">News</a></li>
+                <li id="item-soft"><a href="#" onClick="return itemSoftwareClicked(event);">Software &#9660;</a></li>
+                <li id="item-repos"><a href="#" onClick="return itemRepositoriesClicked(event);">Repositories &#9660;</a></li>
+                <li id="item-support"><a href="#" onClick="return itemSupportClicked(event);">Support &#9660;</a></li>
+                <li id="item-community"><a href="#" onClick="return itemCommunityClicked(event);">Community &#9660;</a></li>
+                <li id="item-donate"><a href="/Donations">Donations</a></li>
+                <li id="item-downloads"><a href="/Downloads">Downloads</a></li>
+                <li id="item-paste"><a href="/Paste">Paste</a></li>
+            </ul>
             </div>
         </div>
-        {% endif %}
+    <!-- End: Header -->
+
+    <div id="subheader" class="container_16">
+        <div id="breadcrump" class="grid_12 alpha">
+            <a href="/" title="Home"><img src="../images/home_grey.png" width="16" height="16" alt="Home" /> KXStudio </a>
+                &gt; <a href="/News">News</a>
+        </div>
+    </div>
+
+    <!-- Start: Main Content Area -->
+    <div id="content" class="container_16 content-wrapper">
+        <div id="some-content" class="box box-shadow grid_77 clearfix">
+            <!-- Start: Box header -->
+            <div class="box-header header-tabs">
+                <ul>
+                    <li><a href="#Top">Top</a></li>
+                </ul>
+            </div>
+            <div id="contentSub"></div>
+            <!-- End: Box header -->
+
+            <div class="alpha omega paste_canvas_area">
+                <h1><span class="mw-headline"> KXStudio&nbsp;: News </span></h1>
+<!-- - - - - - - - Begin Content Area - - - - - - - -->
+
+<div style="position: absolute; top: 120px; right: 100px;">
+    <a href="{% url "feed" %}"><img src="../images/rss.png" alt="RSS"/></a>
+</div>
+
         <div id="wrapper">
-{% include "content/sidebar.twig" %}
             <div id="content">
                 {% for type, messages in flash.all | items %}
                 {% for message in messages %}
@@ -59,41 +155,24 @@
                 <div class="clear"></div>
                 $posts.prev_link
                 $posts.next_link
-                {% if posts.paginated and posts.pages > 1 %}
-                <span class="pages">${ "Page %d of %d" | translate | format(posts.page, posts.pages) }</span>
-                {% endif %}
-                <div id="footer">
-                    <p>Stardust theme &copy; 2008 <a href="mailto:aa.macdonald@gmail.com">Aaron MacDonald</a></p>
-                    <p>${ "Powered by <a class=\"chyrp\" href=\"http://chyrp.net/\">Chyrp</a>" | translate }</p>
-                    <a class="atom" href="{% url "feed" %}">atom</a>
-                </div>
             </div>
             <div class="clear"></div>
         </div>
-        {% if debug and visitor.group.name == "Admin" %}
-        <div id="debug">
-            <h5>
-                <a class="toggle" href="javascript:void(0)">${ "Toggle" | translate }</a>
-                ${ "Stats" | translate }
-            </h5>
+
+<!-- - - - - - - - End Content Area - - - - - - - -->
+            </div>
+            <br/>
+            <div class="box-footer header-tabs">
             <ul>
-                <li><strong>${ "Queries:" | translate }</strong> $sql_queries</li>
-                <li><strong>${ "Load Time:" | translate }</strong> $theme.load_time ${ "seconds" | translate }</li>
+                    <li><a href="#Top">Top</a></li>
             </ul>
-            <br />
-            <h5>${ "Queries" | translate }</h5>
-            <ol class="queries">
-                {% for debug in sql_debug %}
-                <li>
-                    <pre>${ debug.query | replace("\t", "    ") | escape }</pre>
-                    <em>${ "Called from <strong>%s</strong> on line <strong>%d</strong> at %ss." | translate | format(debug.file, debug.line, debug.time) }</em>
-                </li>
-                {% endfor %}
-            </ol>
-            <div class="clear"></div>
         </div>
-        {% endif %}
-${ trigger.call("end_content") }
-        <!-- Queries: $sql_queries | Load Time: $theme.load_time -->
+        </div>
+    </div>
+    <!-- End: Main Content Area -->
+
+    <!-- Note: this clears floating, set in previous elements -->
+    <div class="clear"></div>
+    <div id="footer"></div>
     </body>
 </html>
diff -U 3 -H -w -E -d -r -N -- chyrp/themes/stardust/pages/view.twig News/themes/stardust/pages/view.twig
--- chyrp/themes/stardust/pages/view.twig	2011-03-31 11:32:50.000000000 +0200
+++ News/themes/stardust/pages/view.twig	2013-03-17 02:23:03.000000000 +0100
@@ -6,7 +6,6 @@
                 <div class="clear"></div>
 {% include "feathers/" ~ post.feather ~ ".twig" %}
 {% if enabled_modules.comments and (post.comment_count > 0 or post.commentable) %}
-                <br />
                 <h2 class="comments">${ "Comments" | translate }</h2>
                 <ol class="comments" id="comments_$post.id">
                     {% paginate site.comments_per_page comments in post.comments as comment %}
@@ -26,5 +25,6 @@
                 {% if post.comment_status == "closed" %}
                 ${ "Comments are not accepted on this entry." | translate }
                 {% endif %}
+                <br />
 {% endif %}
 {% endblock %}
diff -U 3 -H -w -E -d -r -N -- chyrp/themes/stardust/stylesheets/screen.css News/themes/stardust/stylesheets/screen.css
--- chyrp/themes/stardust/stylesheets/screen.css	2011-03-31 11:32:50.000000000 +0200
+++ News/themes/stardust/stylesheets/screen.css	2013-12-16 23:54:06.000000000 +0100
@@ -1,734 +1,12 @@
-body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td { margin: 0; padding: 0; }
-table { border-collapse: collapse; border-spacing: 0; }
-fieldset,img { border: 0; }
-address,caption,cite,code,dfn,em,strong,th,var { font-style: normal; font-weight: normal; }
-li { list-style: none; }
-caption,th { text-align: left; }
-h1,h2,h3,h4,h5,h6 { font-size: 100%; font-weight: normal; }
-abbr,acronym { border: 0; font-variant: normal; }
-input,textarea,select { font-family: inherit; font-size: inherit; font-weight: inherit; }
-a { text-decoration: none; color: inherit; }
-
-html {
-    font-size: 62.5%;
-    height: 100%;
-}
-body {
-    background-color: #fff;
-    color: #111;
-    font: normal 1.1em/1.5em 'Helvetica Neue', Helvetica, Arial, Verdana, sans-serif;
-    margin: 0;
-    height: 100%;
-}
-
-pre {
-    font-size: 10px;
-}
-code {
-    font-family: monospace;
-}
-pre code {
-    display: block;
-    background: #111;
-    color: #fff;
-    padding: .5em .75em .35em;
-    margin-bottom: 1em;
-}
-
-em {
-    font-style: italic;
-}
-strong {
-    font-weight: bold;
-}
-
-a:link,
-a:visited {
-    color: #222;
-}
-a:hover {
-    text-decoration: underline;
-}
-
-h1,
-h2 {
-    letter-spacing: -.01em;
-}
-h1 {
-    color: #344469;
-    font: bold 2.6em "palatino linotype", palatino, georgia, "times new roman", times, serif;
-    margin-bottom: .3em
-}
-h2 {
-    color: #664761;
-    font: bold 2em "palatino linotype", palatino, georgia, "times new roman", times, serif;
-    margin-bottom: .4em;
-}
-h3 {
-    font: bold 1.75em "palatino linotype", palatino, georgia, "times new roman", times, serif;
-    margin-bottom: .5em;
-}
-h4 {
-    font-size: 1.5em;
-    font-weight: bold;
-    margin-bottom: .9em;
-}
-h5 {
-    font-size: 1em;
-    font-weight: bold;
-    margin-bottom: 1em;
-}
-h6 {
-    font-size: .8em;
-    font-weight: bold;
-    margin-bottom: 1.25em;
-}
-
-ul {
-    margin: 0 0 0 1.75em;
-}
-ul li {
-    list-style-type: disc;
-}
-ol {
-    padding-left: 2em;
-    margin-bottom: 1em;
-}
-ol li {
-    list-style: decimal;
-}
-
-hr {
-    border: 0;
-    border-top: 1px solid #ddd;
-}
-
-a.comment_delete_link,
-a.comment_edit_link,
-a.delete_link,
-a.edit_link,
-a.next_page,
-a.prev_page,
-a.next_post,
-a.prev_post,
-a.cancel,
-a.logout {
-    font-weight: bold;
-}
-
-a.comment_delete_link,
-a.comment_edit_link {
-    float: right;
-    margin: 0 0 0 .5em;
-}
-a.delete_link {
-    color: #d8583e !important;
-}
-a.edit_link {
-    color: #93ba52 !important;
-}
-a.next_page,
-a.prev_page,
-a.next_post,
-a.prev_post {
-    background-color: #e2e2e2;
-    color: #3C4D69;
-    margin: 0 0 2em;
-    padding: .4em .6em;
-}
-a.next_page:hover,
-a.prev_page:hover,
-a.next_post:hover,
-a.prev_post:hover {
-    color: #4E6B83;
-}
-a.cancel {
-    color: #e35533;
-    border: none;
-    text-decoration: underline;
-}
-a.logout {
-    color: #5B7C99;
-    float: right;
-}
-
-label {
-    color: #777;
-    display: block;
-    font: bold 1.4em "palatino linotype", palatino, georgia, "times new roman", times, serif;
-    font-weight: bold;
-    padding: 0 0 0.2em 0;
-}
-input[type="text"],
-input[type="password"],
-textarea {
-    background-color: #f9f9f9;
-    border: 1px solid #ccc;
-    border-bottom: 1px solid #eee;
-    border-right: 1px solid #eee;
-    font-family: 'Monaco', Courier, monospace;
-    margin-bottom: 1.5em;
-    padding: .3em .4em;
-}
-input[type="file"] {
-    margin-bottom: 1.5em;
-}
-input[type="text"],
-input[type="password"] {
-    width: 20em;
-}
-textarea.wide {
-    width: 432px;
-}
-input.big {
-    font-size: 1.5em;
-}
-
-div#controls {
-    background-color: #131313;
-    border-bottom: .1em solid #0f0f0f;
-    padding: .2em 0 .3em;
-    width: 100%;
-}
-div#controls ul {
-    margin: 0 auto;
-    padding: .4em;
-    width: 660px;
-}
-div#controls ul li {
-    background: none;
-    display: inline;
-    font-weight: bold;
-    margin: 0;
-    padding: 0;
-}
-div#controls ul li a:link,
-div#controls ul li a:visited {
-    color: #bbb;
-    border: none;
-    padding: .1em .6em;
-}
-div#controls ul li a:hover {
-    color: #fff;
-}
-div#controls ul li.close a:link,
-div#controls ul li.close a:visited {
-    color: #e35533;
-}
-div#controls ul li.close a:hover {
-    color: #ff6039;
-}
-
-div#wrapper {
-    width: 660px;
-    margin: 2em auto;
-    padding: 0 0 1em 0;
-}
-
-div#content {
-    width: 440px;
-    float: left;
-}
-
-div#sidebar {
-    background: url('../images/dots-grey.gif') top left repeat;
-    float: right;
-    margin: 0 0 2em 0;
-    padding: 1.5em 1em;
-    width: 165px;
-}
-div#sidebar a:link,
-div#sidebar a:visited {
-    padding: .1em .2em;
-    margin: 0 0 0 -.4em;
-}
-div#sidebar a:hover {
-    background-color: #111;
-    color: #ddd;
-    text-decoration: none;
-}
-div#sidebar h1 {
-    color: #664761;
-    font-size: 1.4em;
-    font-style: normal;
-    margin: 1em 0 .2em 0;
-}
-div#sidebar input#search {
-    background-color: #fff;
-    border: 0;
-    margin: -5px auto 0;
-    width: 150px;
-    padding: .4em .7em;
-    display: block;
-}
-div#sidebar ul li {
-    color: #555;
-}
-
-
-div#header {
-    background-color: #111;
-    padding: 2em 0 .6em 0;
-    margin: 0 0 3em;
-    width: 100%;
-}
-div#header .header_box {
-    margin: auto;
-    width: 660px;
-}
-div#header h1 {
-    margin-bottom: .2em;
-}
-div#header h1 a:link,
-div#header h1 a:visited,
-div#header h2 {
-    color: #eee;
-    font-family: Georgia;
-    font-style: italic;
-}
-div#header h1 a:hover {
-    color: #fff;
-    text-decoration: none;
-}
-div#header h2 {
-    color: #664761;
-    margin: -.2em 0 0 0;
-}
-
-div#footer {
-    background: url('../images/dots-lightgrey.gif') top left repeat;
-    clear: both;
-    color: #888;
-    margin: 3em 0;
-    padding: .4em .6em;
-}
-div#footer a.chyrp:hover {
-    color: #6189a3;
-}
-div#footer a.atom {
-    background-color: #bbb;
-    border: none;
-    color: #222;
-    margin: -2.5em .4em 0 0;
-    padding: .3em .6em;
-    float: right;
-}
-div#footer a.atom:hover {
-    background-color: #EEA64A;
-}
-
-div.notice, div.message, div.warning, div.date {
-    color: #111;
-    font-weight: bold;
-    margin: 0 0 1em 0;
-    padding: .5em .8em;
-}
-div.notice {
-    background: url('../images/dots-green.gif') repeat top left;
-}
-div.message {
-    background: url('../images/dots-blue.gif') repeat top left;
-}
-div.warning {
-    background: url('../images/dots-red.gif') repeat top left;
-}
-div.date {
-    background: url('../images/dots-lightgrey.gif') repeat top left;
-    cursor: default;
-}
-div.notice .sub, div.message .sub, div.warning .sub {
-    font-size: .9em;
-    color: #888;
-    margin-left: .5em;
-}
-
-h2.comments {
-    margin: 0 0 1.0em 0;
-}
-ol.comments {
-    list-style: none;
-    margin-bottom: 1em;
-    padding: 0;
-    word-wrap: break-word;
-}
-ol.comments li {
-    list-style: none;
-    margin-bottom: 1.5em;
-    width: auto;
-}
-ol.comments li blockquote {
-    background: #111;
-    color: #eee;
-    margin: 0;
-    padding: 1em 1em 0;
-}
-ol.comments li blockquote p {
-    padding: 0 0 1em;
-}
-ol.comments li blockquote a:link,
-ol.comments li blockquote a:visited {
-    color: #aaa;
-}
-ol.comments li blockquote blockquote {
-    padding: 0 1em;
-    color: #aaa;
-}
-ol.comments li cite {
-    background: #141414;
-    color: #666;
-    display: block;
-    font-style: normal;
-    margin: 0;
-    padding: .4em 1.3em;
-}
-ol.comments li cite a:link,
-ol.comments li cite a:visited {
-    color: #bbb;
-}
-ol.comments li cite a.permalink:link,
-ol.comments li cite a.permalink:visited {
-    color: #666;
-}
-ol.comments li ul,
-ol.comments li ol {
-    margin: 0;
-    padding: 0 0 1em 1.5em;
-}
-ol.comments li ul li,
-ol.comments li ol li {
-    margin: 0;
-    padding: 0;
-}
-ol.comments li ol li {
-    list-style-type: decimal;
-}
-
-#add_comment {
-    margin-top: 3em;
-}
-
-div.post {
-    margin: 0 0 3em;
-    word-wrap: break-word;
-}
-div.post a:link,
-div.post a:visited {
-    color: #3f5161;
-}
-div.post h2.title {
-    font-size: 2em;
-    padding: 0 0 .3em 0;
-    margin: 0;
-}
-div.post h2.title a:link,
-div.post h2.title a:visited {
-    color: #222;
-    font-size: .8em;
-    font-style: normal;
-    font-weight: bold;
-}
-div.post h2.title a:hover {
-    color: #000;
-}
-div.post .pinned_text {
-    content: "!";
-    float: left;
-    font-size: 1.5em;
-    font-weight: bold;
-    margin: .15em 0 0 -1em;
-    color: #e35533;
-}
-div.post p,
-div.page p {
-    padding: 0 0 1em;
-    font-size: 1.1em;
-}
-div.post li,
-div.page li {
-    font-size: 1.1em;
-}
-div.post li p,
-div.page li p {
-    font-size: 1em;
-}
-div.post pre,
-div.page pre {
-    padding: 0 0 1.5em;
-}
-div.post blockquote {
-    padding: .5em .8em;
-    background: #f9f9f9;
-    border: .1em solid #eee;
-    color: #333;
-    margin: 0 0 1em 0;
-}
-div.post blockquote p {
-    padding: 0;
-}
-div.post span.info,
-div.post span.controls,
-div.page span.controls {
-    background-color: #111;
-    color: #666;
-    float: left;
-    margin: .6em 0 0 0;
-    padding: .3em .6em;
-}
-div.post span.info a:link,
-div.post span.info a:visited {
-    color: #bbb;
-    padding: 0;
-}
-div.post span.controls,
-div.page span.controls {
-    background-color: #eee;
-    color: #ccc;
-    margin: .6em 0 0 .6em;
-}
-div.post ul {
-    margin-bottom: 1em;
-}
-div.post.quote blockquote {
-    background: none;
-    border: 0;
-    color: #777;
-    font: 2em/1.25em normal palatino linotype, palatino, georgia, 'times new roman', times, serif;
-    margin: 0 0 -.5em 0;
-    padding: 0;
-}
-div.post.quote .source {
-    color: #999;
-    display: block;
-    margin: 1em 0;
-}
-div.post.link h2 a:link,
-div.post.link h2 a:visited {
-    color: #3f5161;
-    font-size: 1.2em;
-    margin: 0 0 -.5em 0;
-    padding: 0;
-    text-decoration: underline;
-}
-div.post.link h2 a:hover {
-    color: #354451;
-}
-div.post.link p {
-    color: #444;
-}
-div.post.chat p {
-    display: inline;
-}
-div.post.chat ul {
-    font-family: "monaco", monospace;
-    font-size: .9em;
-    margin: 0 1em 1em 0;
-    padding: 0;
-    color: #535353;
-}
-div.post.chat ul li {
-    background: none;
-    list-style-type: none;
-    padding: 0;
-}
-div.post.chat li .label {
-    color: #2D3879;
-}
-div.post.chat li.me .label {
-    color: #AB2F26;
-}
-div.post.photo img {
-    border: .3em solid #111;
-}
-div.post.photo img a:link,
-div.post.photo img a:visited {
-    border: none;
-}
-div.post.photo p,
-div.post.audio p,
-div.post.video p {
-    color: #444;
-    margin: .4em 0;
-}
-
-div.page h1 {
-    margin-bottom: .5em;
-    color: #4f3a56;
-    font: bold 2.2em "palatino linotype", palatino, georgia, "times new roman", times, serif;
-    letter-spacing: -.01em;
-}
-div.page span.controls {
-    float: right;
-}
-
-.inline_edit {
-    color: #ddd;
-    padding: 1em;
-    background: #111;
-    position: relative;
-}
-.inline_edit input[type="text"],
-.inline_edit textarea {
-    background-color: #222;
-    border: none;
-    color: #eee;
-    padding: .4em;
-    width: 410px;
-}
-.inline_edit .more_options_link:link {
-    color: #5B7C99;
-    display: block;
-    font-weight: bold;
-    padding: 0 0 .3em 0;
-}
-.inline_edit .more_options {
-    background-color: #1a1a1a;
-    padding: 1em;
-}
-.inline_edit .more_options p {
-    float: left;
-    width: 50%;
-    min-height: 5em;
-    padding-bottom: .5em;
-    margin: 0;
-}
-.inline_edit .more_options p input[type="text"] {
-    width: 90%;
-    margin-bottom: 0;
-}
-.inline_edit .more_options p .sub {
-    font-size: .8em;
-    color: #777;
-}
-.inline_edit .buttons {
-    padding: .5em;
-    background: #1a1a1a;
-    color: #ddd;
-}
-.inline_edit h2 {
-    color: #83A650;
-}
-.post_edit {
-    margin-bottom: 2em;
-}
-
-div.pagination {
-    margin: 0 0 2em 0;
-}
-
-.sub {
-    font-size: .85em;
-    color: #999;
-    font-weight: normal;
-}
-
-button {
-    background-color: #664761;
-    border: 1px solid #795473;
-    border-bottom: 1px solid #593E55;
-    border-right: 1px solid #593E55;
-    color: #ddd;
-    font-size: 1em;
-    font-weight: bold;
-    padding: 5px 8px;
-}
-button:hover {
-    color: #fff;
-    cursor: pointer;
-}
-
-p.lost_pass {
-    margin: 0 0 1em 0;
-}
-
-span.pages {
-    color: #888;
-    font-weight: bold;
-    padding: 0 0.3em;
-}
-span.who,
-span.waiting {
-    background-color: #e2e2e2;
-    font-weight: bold;
-    display: block;
-    margin: 0 0 1em;
-    padding: 4px 6px;
-}
-
-.clear {
-    clear: both;
-}
-.right {
-    float: right;
-}
-.left {
-    float: left;
-}
-
-/* Miscellaneous */
-    /* Emoticons module. */
-    img.smiley {
-        margin: 0 0 -.3em;
-    }
-
-    /* Tags module. */
-    .tag {
-        font-family: "palatino linotype", palatino, georgia, "times new roman", times, serif;
-    }
-    .tags_select {
-        display: inline-block;
-        margin-top: .5em;
-    }
-    .tags_select a {
-        padding: .1em .4em;
-        border: .1em solid #DDDDA8;
-        background: #FFFFCA;
-        text-decoration: none;
-        border-top-width: 0;
-        border-left-width: 0;
-        color: #555;
-    }
-    .tags_select a.tag_added {
-        background: #111;
-        border-top-width: .1em;
-        border-bottom: 0;
-        border-left-width: .1em;
-        border-right: 0;
-        border-color: #000;
-    }
+/*
+ * Title: openSUSE.org stylesheet (Bento-Theme)
+ * Author: Robert Lihm <rlihm@opensuse.org>
+ * Copyright (c) 2009 AUTHORS.txt (http://www.opensuse.org)
+ * Licensed under the MIT (MIT-LICENSE.txt) license.
+ */
 
-/* Debugger */
-#debug {
-    background: #ffffdd;
-    color: #666;
-    padding: .4em .75em .5em .75em;
-    border-top: .1em solid #ddd;
-    position: fixed;
-    bottom: 0;
-    left: 0;
-    right: 0;
-    max-height: 33%;
-    overflow-y: auto;
-    overflow-x: hidden;
-}
-#debug h5 {
-    margin: 0;
-    font-size: 1.1em;
-    color: #a3a39a;
-}
-#debug h5 .sub {
-    color: #b3b3aa;
-}
-#debug ol.queries li {
-    margin: 0 0 1em;
-}
-#debug pre {
-    background: #fff;
-    border: 1px solid #ddd;
-    padding: 0 .5em;
-    word-wrap: break-word;
-}
-#debug .toggle {
-    font-weight: bold;
-    color: #666;
-    float: right;
-}
+@import url("../../../../css/reset.css");	/* reset default browser settings */
+@import url("../../../../css/grid.css"); 	/* add fluid 960 Grid System */
+@import url("../../../../css/base.css");	/* Basic CSS - Common styles */
+@import url("../../../../css/base.fluid.fix.css");	/* Fixes for Bento with fluid 960gs */
+@import url("../../../../css/style.css");
